frontend/src/components/layout/ErrorBoundary.tsx

```diff
@@ -0,0 +1,40 @@
+import Custom404 from "pages/404"
+import React from "react"
+
+interface ErrorBoundaryProps {
+  children: React.ReactNode
+}
+
+interface ErrorBoundaryState {
+  hasError: boolean
+}
+
+class ErrorBoundary extends React.Component<
+  ErrorBoundaryProps,
+  ErrorBoundaryState
+> {
+  constructor(props: ErrorBoundaryProps) {
+    super(props)
+    this.state = { hasError: false }
+  }
+
+  // we can write any error handling logic here (e.g. log to sentry)
+  componentDidCatch(error: Error, errorInfo: React.ErrorInfo): void {}
+
+  static getDerivedStateFromError() {
+    return { hasError: true }
+  }
+
+  render(): React.ReactNode {
+    const { hasError } = this.state
+
+    // render fallback ui
+    if (hasError) {
+      return <Custom404 />
```

Comment:
I think it shouldn't be a 404, maybe use the layout from a 404 page and say that there was an error and maybe a retry button like the docs have? https://nextjs.org/docs/advanced-features/error-handling